### PR TITLE
fix: input changes

### PR DESCRIPTION
**Pull request by SteliosKon**
- Added internal state to handle clear icon conditional rendering
- Added ref to handle focus issues

### CODE REVIEW

**Review by lem8n — APPROVED** [review]

File: src/components/FormElements/Input/Input.tsx [hunk 1]
```diff
@@ -54,9 +60,25 @@ const Input: ForwardRefRenderFunction<HTMLInputElement, InputProps> = (
   });
   const iconHeight = size === "sm" ? 28 : 32;
 
+  useImperativeHandle<HTMLInputElement | null, HTMLInputElement | null>(
+    forwardedRef,
+    () => internalRef.current,
+  );
+
+  const resetFocus = () => {
```
Comment (ThrasyvoulosKafasis) [inline src/components/FormElements/Input/Input.tsx:68]: resetFocus is a little confusing name. Something like setFocus or focusInput would be more appropriate.

File: src/components/FormElements/Input/styles.ts [hunk 2]
```diff
@@ -61,7 +66,7 @@ export const inputContainer = (
 
   &.with-suffix-icon {
     .input-wrapper input {
-      padding-inline-end: 3.2rem;
+      padding-inline-end: 5rem;
```
Comment (ThrasyvoulosKafasis) [inline src/components/FormElements/Input/styles.ts:69]: Apply less padding when input is clearable

**Review by ThrasyvoulosKafasis — CHANGES_REQUESTED** [review]

File: src/components/FormElements/Input/Input.stories.tsx [hunk 3]
```diff
@@ -38,7 +40,19 @@ export default {
   ],
 };
 
-const Template: Story<InputProps> = (args) => <InputComponent {...args} />;
+const Template: Story<InputProps> = (args) => {
+  const [state, setState] = useState("My valu");
```
Comment (ThrasyvoulosKafasis) [inline src/components/FormElements/Input/Input.stories.tsx:44]: Please remove default value

File: src/components/FormElements/Input/Input.tsx (hunk 1 above)
Comment (SteliosKon) [reply src/components/FormElements/Input/Input.tsx]: Fixed [commit](https://github.com/epignosis/gnosis/pull/177/commits/cb6bbee19db49eae1ec9a6f410ee6a5e2e8accaf)

File: src/components/FormElements/Input/styles.ts (hunk 2 above)
Comment (SteliosKon) [reply src/components/FormElements/Input/styles.ts]: Fixed [commit](https://github.com/epignosis/gnosis/pull/177/commits/cb6bbee19db49eae1ec9a6f410ee6a5e2e8accaf)

File: src/components/FormElements/Input/Input.stories.tsx (hunk 3 above)
Comment (SteliosKon) [reply src/components/FormElements/Input/Input.stories.tsx]: Fixed [commit](https://github.com/epignosis/gnosis/pull/177/commits/cb6bbee19db49eae1ec9a6f410ee6a5e2e8accaf)

**Pull request merged**